graph.py

```diff
@@ -140,7 +140,7 @@ def sizeof_fmt(num, suffix='bps'):
 
 def discover_links(graph, this_device):
     global debug_id
-    cursor.execute(query_template, this_device)
+    cursor.execute(query_template % this_device)
```

Comment:
Why this change ?
Using parameter instead of string interpolation protects from potential SQL injections.

Reply:
Because with ',' I get:

Generating graph : Traceback (most recent call last):
  File "/opt/observium/graph.py", line 191, in <module>
    graph = discover_links(graph, this_device)
  File "/opt/observium/graph.py", line 145, in discover_links
    cursor.execute(query_template, this_device)
  File "/usr/lib64/python2.7/site-packages/MySQLdb/cursors.py", line 187, in execute
    query = query % tuple([db.literal(item) for item in args])
TypeError: not all arguments converted during string formatting

I know it is not best practice, but at least it makes this thing work for now. If anyone wants to make a better version, feel free :)

Reply:
Have you tried with `cursor.execute(query_template, str(this_device))` and removing the `'` you added in the SQL query ?
(I don't have any working Observium setup to test)

Reply:
Ok let's do your way until we find a better fix